src/main/java/com/cloudbees/plugins/deployer/engines/Engine.java

```diff
@@ -232,7 +234,7 @@ public void log(String message) {
         throw new DeployException("Deployment hosts of type " + configuration.getClass() + " are unsupported");
     }
 
-    public static class FingerprintDecorator implements Callable<DeployedApplicationLocation, IOException> {
+    public static class FingerprintDecorator extends MasterToSlaveCallable<DeployedApplicationLocation, IOException> {
```

Comment:
@jglick So let's see if I follow you. You mean that:
- The engine.process() is executed in master.
- It calls the wrapper, which is executed in the agent.
- This one calls the decorator, which is executed in the master.
- Again in the wrapper, eventually it returns this information to the master.

And you think such "travel" to decorate could be avoided by returning what's necessary in the wrapper and just using it in the process() method. Is that it?

If so, what it's not clear to me is if you think this is something to improve or something that you think that is required. Because if the first one, I would prefer to tackle that in another PR to move this forward and not to increase the size of this one.